### PR TITLE
# 191 StreamAnalytics-Jobs support

### PR DESCRIPTION
**Pull request by marclerwick**
Added support for:

Stream Analytics Functions
Stream Analytics Inputs
Stream Analytics Jobs
Stream Analytics Outputs
Stream Analytics Transformations

### CODE REVIEW

File: Modules/BenchPress.Azure/Public/Confirm-StreamAnalyticsFunction.ps1 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# end INLINE_SKIP
+
+function Confirm-StreamAnalyticsFunction {
+  <#
+    .SYNOPSIS
+      Confirms that a Stream Analytics Function exists.
+
+    .DESCRIPTION
+      The Confirm-AzBPStreamAnalyticsFunction cmdlet gets a Stream Analytics Function using the specified Resource
+      Group, the name of the Job executging the Function, and the name of the Function.
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-StreamAnalyticsFunction.ps1:14]: Typo: executing

File: examples/StreamAnalytics/StreamAnalytics.Tests.ps1 [hunk 2]
```diff
@@ -1,5 +1,5 @@
 BeforeAll {
-  Import-Module Az-InfrastructureTest
+  Import-Module Az-InfrastructureTesting
```
Comment (msha1026) [inline examples/StreamAnalytics/StreamAnalytics.Tests.ps1:2]: Change the hyphen to a period as well.

File: examples/StreamAnalytics/StreamAnalytics_Example.md [hunk 3]
```diff
@@ -23,11 +25,16 @@
...
+   - `teststreamcluster`  -> `your-stream-analytics-cluster-name`
+   - `testjob`            -> `your-stream-analytics-job-name`
+   - `testfunction`       -> `your-stream-analytics-function-name`
+   - `testinput`          -> `your-stream-analytics-input-name`
+   - `testoutput`         -> `your-stream-analytics-output-name`
+   - `testtransformation` -> `your-stream-analytics-transformation-name`
+   - `westus3`            -> `your-location-of-your-cluster`
 
-1. If using a local copy of `Az-InfrastructureTest`, replace `Import-Module Az-InfrastructureTest` with
+1. If using a local copy of `Az-InfrastructureTesting`, replace `Import-Module Az-InfrastructureTesting` with
```
Comment (msha1026) [inline examples/StreamAnalytics/StreamAnalytics_Example.md:37]: Change hyphen to period as well

**Review by msha1026 — APPROVED** [review]
Other than the small fixes, looks great!

File: Modules/BenchPress.Azure/Public/Confirm-StreamAnalyticsFunction.ps1 (hunk 1 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Confirm-StreamAnalyticsFunction.ps1]: Good catch. Fixed.

File: examples/StreamAnalytics/StreamAnalytics.Tests.ps1 (hunk 2 above)
Comment (marclerwick) [reply examples/StreamAnalytics/StreamAnalytics.Tests.ps1]: Fixed.

File: examples/StreamAnalytics/StreamAnalytics_Example.md (hunk 3 above)
Comment (marclerwick) [reply examples/StreamAnalytics/StreamAnalytics_Example.md]: Fixed.

File: examples/StreamAnalytics/StreamAnalytics.Tests.ps1 [hunk 4]
```diff
@@ -50,3 +50,220 @@ Describe 'Verify Stream Analytics Cluster' {
...
+    $name = 'testjob'
+
+    #act
+    $result = Confirm-AzBPStreamAnalyticsJob -ResourceGroupName $rgName -Name $name
+
+    #assert
+    $result | Should -BeInResourceGroup 'rg-test'
+  }
+
+#######################################################################################################################
```
Comment (ozboms) [inline examples/StreamAnalytics/StreamAnalytics.Tests.ps1:102]: Think we should add this to all the examples. Some of them are really hard to decipher

**Pull request merged**